File: final_frontend/forecast_module/forecast.js
// Make an API request and get the forecast data
// const BASE_URL = 'https://api.openweathermap.org/data/3.0/onecall?lat=48.0706687&lon=-0.7734022&exclude=hourly,minutely,alerts&appid=9b14c40da416807a32337c8ec78d4c15'

  fetch(BASE_URL)
  .then(response => response.json())
  .then(data => {
      // Extract and display the weather for the next 4 days
    const dailyForecasts = data.daily.slice(0, 4); // Get data for the next 4 days (including the current day)

    dailyForecasts.forEach((dayData, index) => {
      const forecastContainer = document.createElement('div');
      forecastContainer.classList.add('forecast-container');

      const dayElement = document.createElement('p');
      dayElement.classList.add('forecast-container__day');
      if (index === 0) {
        dayElement.textContent = 'Today';
      } else {
        dayElement.textContent = getDayFromTimestamp(dayData.dt);
      }
      
      forecastContainer.appendChild(dayElement);

      const iconElement = document.createElement('img');
      iconElement.classList.add('forecast-container__icon');
      iconElement.src = getIconURL(dayData.weather[0].description);
      forecastContainer.appendChild(iconElement);

      const temperatureElement = document.createElement('p');
      temperatureElement.classList.add('forecast-container__temperature');
      temperatureElement.textContent = `${(dayData.temp.day - 273.15).toFixed(1)} °`;
      forecastContainer.appendChild(temperatureElement);

      const rainElement = document.createElement('p');
      rainElement.classList.add('forecast-container__rain');
      rainElement.textContent = `${(dayData.rain || 0).toFixed(1)}`;
      forecastContainer.appendChild(rainElement);

      document.querySelector('.dashbord__module__content--forecast').appendChild(forecastContainer);
    });
  })

    // Function to get the day from a timestamp
    function getDayFromTimestamp(timestamp) {
      const date = new Date(timestamp * 1000);
      return date.toLocaleDateString('en-US', { weekday: 'short' });
    }

    // Function to get the icon URL based on the description
    function getIconURL(description) {
      // Create a mapping of descriptions to icons
      const descriptionToIcon = {
        'moderate rain': 'icon-rain.png',
        'light rain': 'icon-sun_rain.png',
        'heavy intensity rain': 'icon-heavy_rain.png',
        'clear sky': 'icon-sun.png',
        'few clouds': 'icon-few-cloud.png',
        'scattered clouds': 'icon-few-cloud.png',
        'overcast clouds': 'icon-few-cloud.png',
        'broken clouds': 'icon-few-cloud.png',
        'fog': 'icon-fog.png',
        // Add more descriptions and their corresponding icons here
      };

      // Check if the description exists in the mapping
      if (descriptionToIcon.hasOwnProperty(description)) {
        return `./styles/assets/${descriptionToIcon[description]}`;
      } else {
        // Handle the case where the description is not in the mapping or provide a default icon
        return './styles/assets/icon_carrot.png'; // Replace with your default icon
      }
    }

File: final_frontend/forecast_module/forecast.py
from datetime import datetime, timedelta
import requests
from geopy.geocoders import Nominatim

# Your OpenWeatherMap API Key
API_KEY = "9b14c40da416807a32337c8ec78d4c15"
lat = "48.8588897"
lon = "2.320041"
exclude = "hourly,minutely,alerts"

# Calculate the timestamp for the start of the current day
current_day = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
current_day_timestamp = int(current_day.timestamp())

# Calculate the timestamp for the start of the next day
next_day = current_day + timedelta(days=1)
next_day_timestamp = int(next_day.timestamp())

# Define the API endpoint
BASE_URL = f"https://api.openweathermap.org/data/3.0/onecall?lat={lat}&lon={lon}&exclude={exclude}&appid={API_KEY}"

# Make the API request
response = requests.get(BASE_URL)


if response.status_code == 200:
    data = response.json()

    # Reverse geocoding to get city name
    geolocator = Nominatim(user_agent="geoapiExercises")
    location = geolocator.reverse(f"{lat}, {lon}")
    city_name = location.raw.get("address", {}).get("city", "Unknown City")

    # Extract current weather
    current_weather = data["current"]
    current_temperature_kelvin = current_weather["temp"]
    current_temperature_celsius = current_temperature_kelvin - 273.15
    current_description = current_weather["weather"][0]["description"]

    # Print location and current weather
    print(f"Location: {city_name}")
    print(f"Current Temperature: {current_temperature_celsius:.2f} °C")
    print(f"Current Weather: {current_description}\n")

    # Extract and print daily weather for the next few days
    daily_forecasts = data["daily"][:4]
    for day in daily_forecasts:
        date = day["dt"]
        temperature_kelvin = day["temp"]["day"]
        temperature_celsius = temperature_kelvin - 273.15
        description = day["weather"][0]["description"]
        date_str = datetime.fromtimestamp(date).strftime('%Y-%m-%d')
        print(f"Date: {date_str}, Temperature: {temperature_celsius:.2f} °C, Weather: {description}")

else:
    print(f"Failed to retrieve data. Status code: {response.status_code}")